x-pack/plugins/security_solution/public/detections/components/osquery/use_osquery_context_action_item.tsx

```diff
@@ -11,17 +11,26 @@ import { useKibana } from '../../../common/lib/kibana';
 
 interface IProps {
   handleClick: () => void;
+  agentId?: string;
 }
 
-export const useOsqueryContextActionItem = ({ handleClick }: IProps) => {
+export const useOsqueryContextActionItem = ({ handleClick, agentId }: IProps) => {
   const osqueryActionItem = useMemo(
     () => <OsqueryActionItem handleClick={handleClick} />,
     [handleClick]
   );
-  const permissions = useKibana().services.application.capabilities.osquery;
+  const {
+    osquery,
+    application: {
+      capabilities: { osquery: permissions },
+    },
+  } = useKibana().services;
+  const osqueryAvailable = osquery?.isOsqueryAvailable({
```

Comment:
Can you please verify if this function (API calls inside)  is not triggered on the list too many times? Meaning, query per each Event in alerts?  I think that the reason I didn't put it here was mainly that, now can't remember.